pycls/core/trainer.py

```diff
@@ -20,11 +23,35 @@
 import torch
 import torch.cuda.amp as amp
 from pycls.core.config import cfg
+from pycls.core.io import pathmgr
 
 
 logger = logging.get_logger(__name__)
 
 
+def setup_env():
+    """Sets up environment for training or testing."""
+    if dist.is_master_proc():
+        # Ensure that the output dir exists
+        pathmgr.mkdirs(cfg.OUT_DIR)
+        # Save the config
+        config.dump_cfg()
+    # Setup logging
+    logging.setup_logging()
+    # Log torch, cuda, and cudnn versions
+    version = [torch.__version__, torch.version.cuda, torch.backends.cudnn.version()]
+    logger.info("PyTorch Version: torch={}, cuda={}, cudnn={}".format(*version))
+    # Log the config as both human readable and as a json
+    logger.info("Config:\n{}".format(cfg)) if cfg.VERBOSE else ()
+    logger.info(logging.dump_log_data(cfg, "cfg", None))
+    # Fix the RNG seeds (see RNG comment in core/config.py for discussion)
+    np.random.seed(cfg.RNG_SEED)
+    torch.manual_seed(cfg.RNG_SEED)
+    random.seed(cfg.RNG_SEED)
+    # Configure the CUDNN backend
+    torch.backends.cudnn.benchmark = cfg.CUDNN.BENCHMARK
+
+
```

Comment:
There's a reason I moved this into it's own file. Moving it back here will break the ability to run pycls on the fb entirely. Leaving it here would force us to override the trainer.py file entirely on the fb stack. It's much simpler to just override the env function.